build/run-in-docker

```diff
@@ -115,4 +115,4 @@ $DOCKER_CMD run $DOCKER_GPU_OPTS $DOCKER_RUN_EXTRA_ARGS -u $(id -u):$(id -g) --r
   -e TZ=${TZ} \
   $DOCKER_OPTS \
   $JNI_DOCKER_IMAGE \
-  scl enable gcc-toolset-11 "$RUN_CMD"
+  scl enable gcc-toolset-14 "$RUN_CMD"
```

Comment:
Additionally, since version information is currently distributed across several locations (such as rapids, JNI, and internal CI), it would be better to make a single source of truth for each branch.

This would allow different releases to automatically determine the appropriate toolset based on the local HEAD, rather than requiring the version to be hardcoded by any caller

Reply:
Sure, please move forward with this, either pushing to this branch or creating a new one.

Reply:
sure, let me add some commits to your change